info-articles/Mapping in solidity - issue #19.md/article.md

```diff
@@ -0,0 +1,414 @@
+
+Recall that the default value of bool is false. Therefore, we declared the negation of the successful mapping against the key. Then we stated that the compiler should push the key into the array if a successful key is true.
+```
+  function getLength() public view returns(uint){
+        return keys.length;
+      }
+
+This function attempts to get the length of the array.
+
+  function getTheI(uint i) public view returns(uint) {
+        return theBalanceOf[keys[i]];
+      }
+```
+We will need the `i variable` when we want to run a loop. Thus, this function returned the balances of keys that form the i variable. Once you have this, you can loop through the mappings we examined earlier in this guide.
+```
+    function getGenesis() public view returns(uint) {
+        return theBalanceOf[keys[0]];
+      }
+ 
+      function getTheLast() public view returns(uint) {
+       return theBalanceOf[keys[keys.length -1]];
+      }
+```
+These two getter functions retrieved the balances of key[0], which is the array's first element, and the array's last member, which is the length minus 1.
+
+## Creating a Custom Contract in Solidity to Explain Mapping
+Since we have learned much about mapping, let us create a simple contract and use the knowledge in real time.
+```
+// SPDX-License-Identifier: MIT
+ pragma solidity ^0.8.16;
+ 
+  contract TheWallet{
+      address public John;
+ 
+      constructor() {
+          John == msg.sender;
+      }
+   
+    mapping(address => uint) public hisBalance;
+    mapping (address=> mapping(address => uint)) public allowThem;
+   
+    function deposit() public payable {
+        hisBalance[msg.sender] += msg.value;
+    }
+ 
+    function withdrawal() public payable {
+        (bool sent, ) = msg.sender.call{value: hisBalance[msg.sender]}("");
+        require(sent, "cannot withdraw");
+    }
+ 
+    function transfer(address receiver, uint token) public returns(bool){
+        allowThem[msg.sender][receiver] += token;
+        hisBalance[msg.sender] += token;
+        hisBalance[receiver] -= token;
+        return true;
+    }
+ 
+    function remainingMoney() public view returns (uint) {
+        return address(this).balance;
+    }
+  }
+ ```
+ 
+
+This contract is a simple wallet contract where users can deposit, transfer, and know the remaining Ethers in the contract. Here, we used mapping to track both balances and allowance.
+
+The next stage is to compile and deploy this contract using Hardhat. Perhaps you have not installed Hardhat in your terminal; paste this on your terminal:
+```
+nvm install latest
+```
+For the compilation, run this command:
+```
+npx hardhat compile
+```
+Bear in mind that you should not have put on the auto-compile option on Remix. Or else, the CLI will return that there is nothing to compile.
+
+Once that is successful, deploy it to your favorite testnet using the Chainstack Ethereum URL. First, create a file in deploy.js and run this script:

```

Comment:
This is not a deploy script. This is the configuration that goes inside the hardhat.config file. This isn't how a deploy script goes.

Reply:
Changed.